cmd/node_agent/na/nafactory.go

```diff
@@ -41,22 +39,17 @@ func NewNodeAgent(cfg *Config) (NodeAgent, error) {
 		certUtil: CertUtilImpl{},
 	}
 
-	switch cfg.Env {
-	case "onprem":
-		na.pr = &onPremPlatformImpl{cfg.CertChainFile}
-	case "gcp":
-		na.pr = &gcpPlatformImpl{&cred.GcpTokenFetcher{Aud: fmt.Sprintf("grpc://%s", cfg.IstioCAAddress)}}
-	case "aws":
-		na.pr = &awsPlatformImpl{ec2metadata.New(session.Must(session.NewSession()))}
-	default:
-		return nil, fmt.Errorf("Invalid env %s specified", cfg.Env)
+	if pc, err := platform.NewClient(cfg.Env, cfg.PlatformConfig, cfg.IstioCAAddress); err == nil {
+		na.pc = pc
+	} else {
+		return nil, err
 	}
 
 	cAClient := &cAGrpcClientImpl{}
 	na.cAClient = cAClient
 
 	secretServer, err := workload.NewSecretServer(
-		workload.NewSecretFileServerConfig(cfg.CertChainFile, cfg.KeyFile))
+		workload.NewSecretFileServerConfig(cfg.PlatformConfig.CertChainFile, cfg.PlatformConfig.KeyFile))
```

Comment:
Is it right to pass `CertChainFile` and `KeyFile` here? I thought these options are for node agent identity.

Reply:
I also think this is problematic. It only works on single-tenant on-prem VMs, where the node agent identity is the same as the service identity. But for other cases, this should be different. Will add a TODO here.